Feature/TA-5-On-Boarding

PeshoOReda opened the pull request:
## 📌 Pull Request Checklist

### ✅ General
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] The PR title follows the required format: `Feature/TA-5-On-Boarding`.
- [x] The branch name follows the required format: `feature/meaningful-name`, `fix/issue-description`, etc.

### 📂 Code Quality
- [x] All filenames are in `snake_case`.
- [x] All functions and variable names follow Dart naming conventions.
- [x] All new code is formatted with `dart format .`

### 🛠 CI/CD Validation
- [x] Linting passes with `dart analyze`.
- [x] The branch name is validated against the allowed patterns.
- [x] The PR title is validated against the required format.

### 🚀 Testing
- [x] Unit tests are included for all new features and bug fixes.
- [x] `flutter test --coverage` has been run and the results reviewed.
- [x] The project builds successfully without errors.

### 📝 Additional Information
This update enhances the **password recovery process**, ensuring a structured and scalable implementation while improving user experience. Future updates will focus on adding **unit tests** to validate functionality and maintain reliability.

### 🔗 Related JIRA Ticket(s):
- [x] JIRA Issue: `TA-5`

### 📸 Screenshots (if applicable):
<div align="center">

PeshoOReda committed:
feat(onboarding): implement onboarding screen with custom buttons and text fields

PeshoOReda committed:
fix(main): update home screen to use OnBoarding as initial screen

PeshoOReda committed:
feat(devtools): add configuration file for Dart & Flutter DevTools settings

PeshoOReda committed:
no thing

PeshoOReda committed:
feat(onboarding): add onboarding screen with custom buttons and text fields

- Add `EasyLocalization` package
- Implement OnBoarding screen with a logo, title, and text fields
- Create custom button and text field widgets

PeshoOReda committed:
fix(main): Change initial screen to SplashScreen

- Updated `main.dart` to use `SplashScreen` as the initial screen instead of `OnBoarding`.

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>

PeshoOReda committed:
refactor(main): remove unused TODO comment and unimplemented error

The unnecessary TODO comment and the UnimplementedError within the build method of _MyHomePageState have been removed as they were not being used and caused an error.

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>

PeshoOReda committed:
Feature/TA-5-On-Boarding

PeshoOReda committed:
fix(onboarding): implement initial onboarding screen design

- Remove unused custom button and text field widgets.
- Set OnBoardingScreen as the initial screen in main.dart.
- Replace custom buttons with standard ElevatedButton.
- Adjust layout and styling for the onboarding screen.

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>

PeshoOReda committed:
refactor(onboarding): remove unused custom button and text field files

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>

Ahmed-Abdelghany-11 commented on lib/features/onBoarding/presentation/view/on_boarding_screen.dart:
```diff
@@ -1 +1,73 @@
+              style: ElevatedButton.styleFrom(backgroundColor: AppColors.pink),
+              child: Text(
+                tr(LocaleKeys.Login),
+                style: const TextStyle(color: AppColors.white, fontSize: 15),
```
use the app theme for text styling

PeshoOReda replied:
هو مكنوش مناسبين بس هعملهم

Ahmed-Abdelghany-11 commented on lib/features/onBoarding/presentation/view/on_boarding_screen.dart:
```diff
@@ -1 +1,73 @@
+              ),
+              child: Text(
+                tr(LocaleKeys.ApplyNow),
+                style: const TextStyle(color: AppColors.black, fontSize: 15),
```
use the app theme for text styling

Ahmed-Abdelghany-11 commented on lib/features/onBoarding/presentation/view/on_boarding_screen.dart:
```diff
@@ -1 +1,73 @@
+                child: Text(
+                  'Version 1.0.0 (2006)',
+                  textAlign: TextAlign.center,
+                  style: TextStyle(fontSize: 12, color: AppColors.gray),
```
use the app theme for text styling

ali72-20 reviewed (approved):
ahmed comment is perfect follow it

PeshoOReda committed:
refactor(onboarding): update on boarding screen with theme and screenutil

- Applied text themes to screen
- Added ScreenUtil for responsive padding and sizing
- Updated button border radius

Signed-off-by: PeshoOReda <drmafiaemail13@gmail.com>

PeshoOReda committed:
Feature/TA-5-On-Boarding

ahmedelazab1220 committed:
Merge branch 'developement' of github.com:ahmedelazab1220/tracking_flower_app into feature/ta-5-on-boarding

ahmedelazab1220 committed:
refactor(localization): add version key to translations and update onboarding screen layout

ahmedelazab1220 committed:
refactor(locale_keys): format code for better readability
